docker/build_artifacts/sagemaker/serve.py

```diff
@@ -94,25 +94,24 @@ def __init__(self):
             if low + 2 * self._tfs_instance_count > hi:
                 raise ValueError("not enough ports available in SAGEMAKER_SAFE_PORT_RANGE ({})"
                                  .format(self._sagemaker_port_range))
-            self._tfs_grpc_port_range = "{}-{}".format(low,
+            # select non-overlapping grpc and rest ports based on tfs instance count
+            # record the init and end value in order to reconstruct them in python service
+            self._tfs_grpc_selected_ports = "{}-{}".format(low,
                                                        low + 2 * self._tfs_instance_count)
-            self._tfs_rest_port_range = "{}-{}".format(low + 1,
+            self._tfs_rest_selected_ports = "{}-{}".format(low + 1,
                                                        low + 2 * self._tfs_instance_count + 1)
```

Comment:
It might be worth considering a slightly different approach where instead of sending lower and upper bound we could send the actual ports (in a comma-separated fashion or something similar).
That way, there would be no need to reconstruct anything on the receiving end and there would be no need to keep the algorithms in sync (reducing the chance for errors).

Reply:
yeah, good point. Let me modify in next revision.